querybook/server/logic/comment.py

```diff
@@ -0,0 +1,147 @@
+from querybook.server.app.db import with_session
+from models.comment import Comment, CommentReaction, DataTableComment, DataCellComment
+from querybook.server.const.comment import CommentDict
+
+
+@with_session
+def get_comment_by_id(comment_id: int, session=None):
+    return Comment.get(id=comment_id, session=session)
+
+
+@with_session
+def get_all_comment_data_dict_by_id(comment_id: int, session=None) -> CommentDict:
+    comment = Comment.get(id=comment_id, session=session)
+    child_comments = (
+        session.query(Comment)
+        .filter(Comment.parent_comment_id == comment_id)
+        .order_by(Comment.created_at.asc())
+        .all()
+    )
+    reactions = (
+        session.query(CommentReaction)
+        .filter(CommentReaction.comment_id == comment_id)
+        .all()
+    )
+    comment_dict = comment.to_dict()
+    if child_comments:
+        comment_dict["child_comments"] = [
+            child_comment.to_dict() for child_comment in child_comments
```

Comment:
does it also need to populate the reactions for the child comments?

Also from the UI mockup, it seems we only need the count of the child comments first. And get the details when they click the "2 replies"

Reply:
ah that makes sense. let me update that